flexget/plugins/sites/hliang.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import unicode_literals, division, absolute_import
+
+import logging
+import re
+
+from flexget import plugin
+from flexget.event import event
+from flexget.plugins.internal.urlrewriting import UrlRewritingError
+from flexget.utils.soup import get_soup
+
+log = logging.getLogger('hliang')
+
+class UrlRewriteHliang(object):
+    """Hliang urlrewriter."""
+
+    # urlrewriter API
+    def url_rewritable(self, task, entry):
+        url = entry['url']
+        if url.startswith('http://bt.hliang.com/show.php'):
+            return True
+        return False
+
+    # urlrewriter API
+    def url_rewrite(self, task, entry):
+        entry['url'] = self.parse_download_page(entry['url'], task.requests)
+
+    @plugin.internet(log)
+    def parse_download_page(self, url, requests):
+        txheaders = {'User-agent': 'Mozilla/4.0 (compatible; MSIE 5.5; Windows NT)'}
+        try:
+            page = requests.get(url, headers=txheaders)
```

Comment:
Yeah, there is a catch all but this is irrelevant to my point. 
You have two separate operations, request and soup, and each one can fail independently. Having one big block with all poss exceptions is just poor code which is hard to debug and maintain